### PR TITLE
Continued stabilization for the SSH policy ran by the Universal NRP and the Security Baseline module

### PR DESCRIPTION
**Pull request by MariusNi**
## Description

Continued stabilization for the SSH policy ran by the Universal NRP and the Security Baseline module, including bug fixes and improvements. 

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/common/commonutils/SshUtils.c [hunk 1]
```diff
@@ -528,6 +528,7 @@ int CheckSshProtocol(char** reason, void* log)
     else if (EEXIST == (status = CheckLineNotFoundOrCommentedOut(g_sshServerConfiguration, '#', protocol, log)))
     {
         OsConfigLogInfo(log, "CheckSshProtocol: '%s' is found uncommented in %s", protocol, g_sshServerConfiguration);
+        OsConfigCaptureSuccessReason(reason, "%s'%s' is found uncommented in %s", protocol, g_sshServerConfiguration);
```
Comment (simonjaeger) [inline src/common/commonutils/SshUtils.c:531]: nit: Should the last %s be covered in single quotes for consistency with others?

**Review by simonjaeger — APPROVED** [review]
LGTM

File: src/common/commonutils/CommonUtils.h [hunk 2]
```diff
@@ -34,6 +34,19 @@
...
 }\
 
+#define OsConfigCaptureSuccessReason(reason, FORMAT, ...) {\
+    if (NULL != reason) {\
+        FREE_MEMORY(*reason);\
+        *reason = FormatAllocateString(FORMAT, SECURITY_AUDIT_PASS, ##__VA_ARGS__); \
+    }\
+}\
+
+#define OsConfigResetReason(reason) {\
```
[the remark this answers is not in the document]
Comment (MariusNi) [reply src/common/commonutils/CommonUtils.h:44]: Macros are way more performant as they are expanded inline. Esp for small bits of code like this.

File: src/common/commonutils/SshUtils.c (hunk 1 above)
Comment (MariusNi) [reply src/common/commonutils/SshUtils.c:531]: This is by design and yes, it's already consistent. This last string is the SSH server configuration file and simply it looks better there in reasons w/o the '' delimiters which are reserved for values.

Comment (MariusNi) [reply src/common/commonutils/SshUtils.c:531]: " 'Protocol 2' is found uncommented in /etc/ssh/sshd_config"

Same as for the sshd service here:

" The sshd service reports that 'clientaliveinterval' is set to '3600' (that is greater than zero)"

Comment (MariusNi) [reply src/common/commonutils/SshUtils.c:531]: sshd and /etc/ssh/sshd_config do not get quoted, they are the two main pieces there and are always the same

The values (which can change) get quoted and that only to see if there are spaces sneaking after and such

**Review by robertschaedler3 — APPROVED** [review]

**Pull request merged**